timApp/idesupport/utils.py

```diff
@@ -228,6 +232,85 @@ class TIDECourse:
     """
 
 
+def combine_code_blocks(program: str | None, by: str | None,
+                        by_code: str | None, filename: str | None) -> str:
+    """
+    Combine the code blocks.
+
+    :param program: Code block
+    :param by: Code block
+    :param by_code: Code block
+    :return: Combined code blocks
+    """
+    if program is None and by is None and by_code is None:
+        raise RouteException("No code found in the plugin")
+
+    if by is None and by_code is None:
+        return program
```

Comment:
Tänne ei pitäis päästä mutta mypy valittaa silti. Tän vois sieventää esim näin niin menee mypystä läpi
```suggestion
    if by is None and by_code is None:
        if program is None:
            raise RouteException("No code found in the plugin")
        return program
```

Reply:
Korjattu.